cmd/internal/server/core.go

```diff
@@ -26,10 +24,10 @@ import (
 // Opts exposes additional configuration for server operation.
 type Opts struct {
 	CPUProfilePath      string
-	ExternalPluginsPath string
+	ExternalPlugins     []plugin.ExternalPluginSpec
 	LogFile             string
 	// LogLevel can be "warn", "info", "debug", or "trace".
-	LogLevel string
+	LogLevel            string
```

Comment:
I think we should include the `Socket` as an option, or make it a parameter to `New`. That would remove the config dependency and make the `Server` object more generic.

Reply:
That makes sense.